tests/ui/unnecessary_cast_fixable.rs

```diff
@@ -12,12 +12,19 @@ fn main() {
     #[rustfmt::skip]
     let v = vec!(1);
     &v as &[i32];
-    1.0 as f64;
-    1 as u64;
     0x10 as f32;
     0o10 as f32;
     0b10 as f32;
     0x11 as f64;
     0o11 as f64;
     0b11 as f64;
+
+    1 as u32;
+    0x10 as i32;
+    0b10 as usize;
+
```

Comment:
Negative literals are not linted (e.g. `-1 as i32`). I think we could improve that, but since that behavior existed in the previous implementation of the lint, I will leave that up to you :)

If you want to address it, we should cover casts where the "from" expression is an unary `UnOp::UnNeg` operation on a literal.

Reply:
I'm on it

Reply:
Done